crypto/types/types.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+package cosmos_sdk.crypto.v1;
+
+import "third_party/proto/gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/types";
+
+// PublicKey specifies a public key
+message PublicKey {
+    // sum specifies which type of public key is wrapped
+    oneof sum {
+        bytes secp256k1 = 1;
+        bytes ed25519 = 2;
+        bytes sr25519 = 3;
+        PubKeyMultisigThreshold multisig = 4;
+        bytes secp256r1 = 5;
+
+        // any_pubkey can be used for any pubkey that an app may use which is
+        // not explicitly defined in the oneof
+        google.protobuf.Any any_pubkey = 15;
```

Comment:
How would this work for an external application? Eg: on Ethermint we use ethereum's secp256k1 implementation: 

https://github.com/ChainSafe/ethermint/blob/542a25ff827036fe9305b154f23ec86f45f48947/crypto/secp256k1.go#L78-L80

It'd be nice to have a concrete example to ease the migration

Reply:
I'm thinking there would be two ways to handle this.

One is we don't define the mapping to `crypto.PubKey` on this type and instead let that be a parameter for `TxDecoder`, i.e. `func (types.PublicKey) (crypto.PubKey, error)`.

Or, you could use `any_pubkey` which would specify the full URL (ex. `ethermint.crypto.PubKeySecp256k1`).